### PR TITLE
Implementation of 'User Permissions' approach

### PR DESCRIPTION
**Pull request by TarasOliinyk**

### CODE REVIEW

File: src/main/java/com/springboot/practice/service/implementation/UserServiceImpl.java [hunk 1]
```diff
@@ -29,13 +34,36 @@ public UserDTO createUser(UserDTO userDTO) {
...
     public UserDTO getUserByUsername(String username) {
         logger.info("Get user with username " + username);
-        return modelMapper.map(userRepository.findByUsername(username).orElseThrow(
+        return modelMapper.map(userRepository.findOneByUsername(username).orElseThrow(
                 () -> new UserNotFoundException("There is no user with username " + username)), UserDTO.class);
     }
+
+    @Override
+    public UserDTO updateUser(UserDTO userDTO) {
+        logger.info(String.format("Update user with id %s, updated user: %s", userDTO.getId(), userDTO.toString()));
```
Comment (geruch13) [inline src/main/java/com/springboot/practice/service/implementation/UserServiceImpl.java:65]: before update check the user

Comment (TarasOliinyk) [reply src/main/java/com/springboot/practice/service/implementation/UserServiceImpl.java]: Done

File: src/main/java/com/springboot/practice/service/implementation/RoleServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public RoleDTO updateRole(RoleDTO roleDTO) {
+        logger.info(String.format("Update role with id %s, updated role: %s", roleDTO.getId(), roleDTO.toString()));
+        Role role = modelMapper.map(roleDTO, Role.class);
+        return modelMapper.map(roleRepository.save(role), RoleDTO.class);
+    }
+
+    @Override
+    public void deleteRole(Integer roleId) {
+        logger.info("Delete role with id " + roleId);
+        roleRepository.deleteById(roleId);
```
[the remark this answers is not in the document]
Comment (TarasOliinyk) [reply src/main/java/com/springboot/practice/service/implementation/RoleServiceImpl.java]: Done

File: src/main/java/com/springboot/practice/service/implementation/PermissionServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        logger.info(String.format("Update permission with id %s, updated permission %s", permissionDTO.getId(),
+                permissionDTO.toString()));
+        Permission permission = modelMapper.map(permissionDTO, Permission.class);
+        return modelMapper.map(permissionRepository.save(permission), PermissionDTO.class);
+    }
+
+    @Override
+    public void deletePermission(Integer permissionId) {
+        logger.info("Delete permission with id " + permissionId);
+        permissionRepository.deleteById(permissionId);
```
[the remark this answers is not in the document]
Comment (TarasOliinyk) [reply src/main/java/com/springboot/practice/service/implementation/PermissionServiceImpl.java]: Done

File: src/main/java/com/springboot/practice/controller/UserController.java [hunk 4]
```diff
@@ -1,23 +1,52 @@
...
     public ResponseEntity<UserDTO> singUp(@RequestBody UserDTO userDTO) {
         return ResponseEntity.status(HttpStatus.CREATED).body(userService.createUser(userDTO));
     }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<UserDTO> getUser(@PathVariable (name = "id") Integer id) {
+        return ResponseEntity.status(HttpStatus.FOUND).body(userService.getUserById(id));
+    }
+
+    @GetMapping("/list")
```
[the remark this answers is not in the document]
Comment (TarasOliinyk) [reply src/main/java/com/springboot/practice/controller/UserController.java:43]: Modified

**Pull request merged**